js/privatebin.js

```diff
@@ -1195,8 +1195,17 @@ jQuery.PrivateBin = (function($, sjcl, Base64, RawDeflate) {
          */
         function sendToShortener()
         {
-            window.location.href = $shortenButton.data('shortener') +
-                                   encodeURIComponent($pasteUrl.attr('href'));
+            var url = $shortenButton.data('shortener')+encodeURIComponent($pasteUrl.attr('href'));
+
+            if($shortenButton.data('ajax'))
+                $.get(url, function(d) {
+                    $pasteUrl.attr(d);
+                    $pasteUrl.html(d);
```

Comment:
No, rather not allow the remote server to inject any HTML here. (It just returns text, does not it?)
Please change it to text-only.